src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -332,17 +367,27 @@ def update_coords(self):
         This is one of four functions affecting the coordinate system visualisation which should be updated if
         a new 3D rendering library is used: `setup_display()`, `update_coords()`, `update_polarisation_coords()`, `set_polarisation_visible()`.
         """
-        if self.txtEnvYaw.hasAcceptableInput() and self.txtEnvPitch.hasAcceptableInput() and self.txtEnvRoll.hasAcceptableInput() \
-           and self.txtSampleYaw.hasAcceptableInput() and self.txtSamplePitch.hasAcceptableInput() and self.txtSampleRoll.hasAcceptableInput():
+        if self.txtEnvYaw.hasAcceptableInput() and self.txtEnvPitch.hasAcceptableInput() and \
+                self.txtEnvRoll.hasAcceptableInput() and self.txtSampleYaw.hasAcceptableInput() and \
+                self.txtSamplePitch.hasAcceptableInput() and self.txtSampleRoll.hasAcceptableInput():
+
             UVW_to_uvw, UVW_to_xyz = self.create_rotation_matrices()
             basis_vectors = numpy.array([[1,0,0],[0,1,0],[0,0,1]])
+
             #TODO: when scipy version updated can just use Rotation.as_matrix() to get new basis vectors - function name currently varies between versions used.
             uvw_matrix = UVW_to_uvw.apply(basis_vectors)
+
             xs, ys, zs = numpy.transpose(numpy.stack((numpy.zeros_like(uvw_matrix), uvw_matrix)), axes=(2, 1, 0))
-            self.coord_arrows[1].update_data(zs, xs, ys) # stack in order zs, xs, ys to match the coord system used in sasview
+            for i in range(3):
```

Comment:
This reflects a bigger change in structure, driven by the need to call Arrow3D for each arrow, rather than just once for all of them. I found this actually draws them, unlike the existing method. The underlying reason for why this works and the existing code didn't - I'm not sure - but I expect its because of a change in how Arrow3D works on the matplotlib side.